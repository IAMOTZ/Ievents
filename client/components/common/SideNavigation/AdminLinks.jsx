import React from 'react';
import { Link } from 'react-router-dom';
import FakeDiv from '../../hoc/FakeDiv';

const AdminLinks = (props) => {
  let component;
  if (props.isAdmin || props.isSuperAdmin) {
    component = (
      <FakeDiv>
        <Link to="/addCenter" className="list-group-item">
          <i className="fa fa-plus fa-fw" aria-hidden="true" />&nbsp; Add Center
        </Link>
        {
          props.isSuperAdmin ?
            <Link to="/addAdmin" className="list-group-item">
              <i className="fa fa-user fa-fw" aria-hidden="true" />&nbsp; Add Admin
            </Link> : null
        }
        <Link to="/transactions" className="list-group-item">
          <i className="fa fa-tasks fa-fw" aria-hidden="true" />&nbsp; Transactions
        </Link>
      </FakeDiv>
    );
  } else {
    component = null;
  }
  return component;
};

export default AdminLinks;
