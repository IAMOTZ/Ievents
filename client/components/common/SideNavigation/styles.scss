@import '../../../styles/variables';

#side-navigation {
  background-color: $custom-black;
  min-height: 2000px;
  .navigation-links {
    a {
      color: white;
    }
    .list-group-item {
      background-color: transparent;
      border-top: 1px solid $custom-blue;
    }
    .list-group-item:last-child {
      border-bottom: 1px solid $custom-blue;
    }
  }
}