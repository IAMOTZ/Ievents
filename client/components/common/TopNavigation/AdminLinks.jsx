import React from 'react';
import { Link } from 'react-router-dom';
import FakeDiv from '../../hoc/Aux';

const AdminLinks = (props) => {
  let component;
  if (props.isAdmin || props.isSuperAdmin) {
    component = (
      <FakeDiv>
        <li className="nav-item">
          <Link to="/addCenter" className="nav-link">Add Center</Link>
        </li>
        {
          props.isSuperAdmin ?
            <li className="nav-item">
              <Link to="/addAdmin" className="nav-link">Add Admin</Link>
            </li> : null
        }
        <li className="nav-item">
          <Link to="/transactions" className="nav-link">Transactions</Link>
        </li>
      </FakeDiv>
    );
  } else {
    component = null;
  }
  return component;
};

export default AdminLinks;
