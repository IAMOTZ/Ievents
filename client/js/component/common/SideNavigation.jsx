import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { clearUser } from '../../actions/authAction';

@connect((store) => {
  return {

  }
})

// This is the side navigation that appears when a user is logged in(Large Screen)
class UserSideNav extends React.Component {
  logout = () => {
    this.props.dispatch(clearUser());
  }
  render() {
    return (
      <div class="col-lg-2 fixed-top  d-none d-lg-block" id="navigation-section">
        <div class="mt-5 text-center">
          <p class="lead text-white pt-3 text-capitalize" id="userName">{this.props.userName}</p>
        </div>
        <div class="pt-3 navigation-links">
          <div class="list-group">
            <Link to='/events' class="list-group-item">
              <i class="fa fa-user-circle fa-fw" aria-hidden="true"></i>&nbsp; My Events
            </Link>
            <Link to="addEvent" class="list-group-item">
              <i class="fa fa-plus fa-fw" aria-hidden="true"></i>&nbsp; Add Events
            </Link>
            <Link to="#" class="list-group-item">
              <i class="fa fa-bank fa-fw" aria-hidden="true"></i>&nbsp; Centers
            </Link>
            <Link to="/" class="list-group-itsem" onClick={this.logout}>
              <i class="fa fa-power-off fa-fw" aria-hidden="true"></i>&nbsp; Logout
            </Link>
          </div>
        </div>
      </div >
    );
  }
}

// This is the side navigation that appears when an admin is logged in(Large Screen)
const AdminSideNav = (props) => {
  return (
    <div class="col-lg-2 fixed-top  d-none d-lg-block" id="navigation-section">
      <div class="profile-pic mt-5 text-center">
        <p class="lead text-white pt-3" id="userName">Admin</p>
      </div>
      <div class="pt-3 navigation-links">
        <div class="list-group">
          <Link to="#" class="list-group-item">
            <i class="fa fa-bank fa-fw" aria-hidden="true"></i>&nbsp; Centers
        </Link>
          <Link to="#" class="list-group-item">
            <i class="fa fa-plus fa-fw" aria-hidden="true"></i>&nbsp; Add Center
        </Link>
          <Link to="#" class="list-group-item">
            <i class="fa fa-tasks fa-fw" aria-hidden="true"></i>&nbsp; Transactions
        </Link>
          <Link to="#" class="list-group-item">
            <i class="fa fa-power-off fa-fw" aria-hidden="true"></i>&nbsp; Logout
        </Link>
        </div>
      </div>
    </div>
  );
};

export { UserSideNav, AdminSideNav };
