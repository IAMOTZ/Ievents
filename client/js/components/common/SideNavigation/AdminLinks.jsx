import React from 'react';
import { Link } from 'react-router-dom';

const AdminLinks = (props) => {
  let component;
  if (props.isAdmin || props.isSuperAdmin) {
    component = (
      <div>
        <Link to="/addCenter" className="list-group-item">
          <i className="fa fa-plus fa-fw" aria-hidden="true" />&nbsp; Add Center
        </Link>
        {
          props.isSuperAdmin ?
            <Link to="/addAdmin" className="list-group-item">
              <i className="fa fa-user fa-fw" aria-hidden="true" />&nbsp; Add Admin
            </Link> : null
        }
        <Link to="/transactions" className="list-group-item">
          <i className="fa fa-tasks fa-fw" aria-hidden="true" />&nbsp; Transactions
        </Link>
      </div>
    );
  } else {
    component = null;
  }
  return component;
};

export default AdminLinks;
