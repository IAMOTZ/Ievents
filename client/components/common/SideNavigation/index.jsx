import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './styles.scss';
import AdminLinks from './AdminLinks';
import { logOut } from '../../../actions/authAction';

const SideNavigation = props => (
  <div className="col-lg-2 fixed-top  d-none d-lg-block" id="side-navigation">
    <div className="mt-5 text-center">
      <p className="lead text-white pt-3 text-capitalize" id="name">{props.name}</p>
    </div>
    <div className="pt-3 navigation-links">
      <div className="list-group">
        <Link to="/events" className="list-group-item">
          <i className="fa fa-book fa-fw" aria-hidden="true" />&nbsp; My Events
        </Link>
        <Link to="addEvent" className="list-group-item">
          <i className="fa fa-plus fa-fw" aria-hidden="true" />&nbsp; Create Event
        </Link>
        <Link to="/centers2" className="list-group-item">
          <i className="fa fa-bank fa-fw" aria-hidden="true" />&nbsp; Centers
        </Link>
        <AdminLinks isAdmin={props.isAdmin} isSuperAdmin={props.isSuperAdmin} />
        <Link to="/profile" className="list-group-item">
          <i className="fa fa-user-circle fa-fw" aria-hidden="true" />&nbsp; Profile
        </Link>
        <Link to="/" className="list-group-item" onClick={() => props.dispatch(logOut())}>
          <i className="fa fa-power-off fa-fw" aria-hidden="true" />&nbsp; Logout
        </Link>
      </div>
    </div>
  </div >
);

SideNavigation.propTypes = {
  name: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default SideNavigation;
