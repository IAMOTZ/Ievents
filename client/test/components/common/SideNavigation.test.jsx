import React from 'react';
import SideNavigation from '../../../components/common/SideNavigation';
import AdminLinks from '../../../components/common/SideNavigation/AdminLinks';

const props = {
  name: 'name',
  isAdmin: false,
  isSuperAdmin: false,
  dispatch: jest.fn(),
};

const alterProps = newProps => ({ ...props, ...newProps });

describe('<SideNavigation />', () => {
  describe('Rendering:', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<SideNavigation {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Interaction', () => {
    it('should dispatch the logout action', () => {
      const wrapper = shallow(<SideNavigation {...props} />);
      wrapper.find('Link[to="/"]').simulate('click');
      expect(props.dispatch.mock.calls.length).toEqual(1);
    });
  });
});

describe('<AdminLinks />', () => {
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<AdminLinks {...props} />);
    });
    it('should render correctly if the user is not an admin', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly if the user is an admin', () => {
      wrapper.setProps(alterProps({ isAdmin: true }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly if the user is a super admin', () => {
      wrapper.setProps(alterProps({ isSuperAdmin: true }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
