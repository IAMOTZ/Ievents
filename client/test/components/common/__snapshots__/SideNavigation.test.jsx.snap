// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminLinks /> Rendering should render correctly if the user is a super admin 1`] = `
<Aux>
  <Link
    className="list-group-item"
    replace={false}
    to="/addCenter"
  >
    <i
      aria-hidden="true"
      className="fa fa-plus fa-fw"
    />
      Add Center
  </Link>
  <Link
    className="list-group-item"
    replace={false}
    to="/addAdmin"
  >
    <i
      aria-hidden="true"
      className="fa fa-user fa-fw"
    />
      Add Admin
  </Link>
  <Link
    className="list-group-item"
    replace={false}
    to="/centers/transactions"
  >
    <i
      aria-hidden="true"
      className="fa fa-tasks fa-fw"
    />
      Transactions
  </Link>
</Aux>
`;

exports[`<AdminLinks /> Rendering should render correctly if the user is an admin 1`] = `
<Aux>
  <Link
    className="list-group-item"
    replace={false}
    to="/addCenter"
  >
    <i
      aria-hidden="true"
      className="fa fa-plus fa-fw"
    />
      Add Center
  </Link>
  <Link
    className="list-group-item"
    replace={false}
    to="/centers/transactions"
  >
    <i
      aria-hidden="true"
      className="fa fa-tasks fa-fw"
    />
      Transactions
  </Link>
</Aux>
`;

exports[`<AdminLinks /> Rendering should render correctly if the user is not an admin 1`] = `""`;

exports[`<SideNavigation /> Rendering: should render correctly 1`] = `
<div
  className="col-lg-2 fixed-top  d-none d-lg-block"
  id="side-navigation"
>
  <div
    className="mt-5 text-center"
  >
    <p
      className="lead text-white pt-3 text-capitalize"
      id="name"
    >
      name
    </p>
  </div>
  <div
    className="pt-3 navigation-links"
  >
    <div
      className="list-group"
    >
      <Link
        className="list-group-item"
        replace={false}
        to="/events"
      >
        <i
          aria-hidden="true"
          className="fa fa-book fa-fw"
        />
          My Events
      </Link>
      <Link
        className="list-group-item"
        replace={false}
        to="/centers"
      >
        <i
          aria-hidden="true"
          className="fa fa-bank fa-fw"
        />
          Centers
      </Link>
      <AdminLinks
        isAdmin={false}
        isSuperAdmin={false}
      />
      <Link
        className="list-group-item"
        replace={false}
        to="/profile"
      >
        <i
          aria-hidden="true"
          className="fa fa-user-circle fa-fw"
        />
          Profile
      </Link>
      <Link
        className="list-group-item"
        onClick={[Function]}
        replace={false}
        to="/"
      >
        <i
          aria-hidden="true"
          className="fa fa-power-off fa-fw"
        />
          Logout
      </Link>
    </div>
  </div>
</div>
`;
